src/test/unit/parsers/test_utils.py

```diff
@@ -211,6 +211,34 @@ def test_recursive_json_join(d1, d2, result):
                 "detailType": "Scheduled Event",
             },
         ),
+        (
+            {
+                "id": "cdc73f9d-aea9-11e3-9d5a-835b769c0d9c",
+                "detail-type": "Unknown",
+                "source": "aws.events",
+                "time": "1970-01-01T00:00:00Z",
+                "region": "us-east-1",
+                "resources": ["arn:aws:events:us-east-1:123456789012:rule/ExampleRule"],
+                "detail": {},
+            },
+            {"triggeredBy": "unknown"},
+        ),
+        (
+            {
+                "id": "cdc73f9d-aea9-11e3-9d5a-835b769c0d9c",
+                "detail-type": "Scheduled Event",
+                "source": "aws.events",
+                "time": "1970-01-01T00:00:00Z",
+                "region": "us-east-1",
+                "detail": {},
+            },
+            {
+                "triggeredBy": "cloudwatch",
+                "resource": "unknown",
+                "region": "us-east-1",
+                "detailType": "Scheduled Event",
+            },
+        ),
```

Comment:
please add another test that checks the `resource` extraction,
preferable get a real data (instead of create it artificially)